### PR TITLE
Epmctdm 7033 multicore for utility metric

### PR DESCRIPTION
**Pull request by Ijka**
New able to use multi CPU core algorithms for utility metric calculation provided. 
Stratified sampling for big datasets for calculations of utility metrics is made

### CODE REVIEW

File: src/syngen/ml/metrics/metrics_classes/metrics.py [hunk 1]
```diff
@@ -1145,35 +1169,69 @@ def __model_process(self, model_object, targets, task_type):
...
         best_target, score, synthetic_score = self.__model_process(
-            LogisticRegression(random_state=10), binary_targets, "binary classification"
+            LogisticRegression(n_jobs=-1, random_state=10),
+            binary_targets, "binary classification"
         )
         return best_target, score, synthetic_score
 
     def __create_multi_class_models(self, multiclass_targets):
-        from sklearn.ensemble import GradientBoostingClassifier
+        from sklearn.ensemble import RandomForestClassifier
```
Comment (Anna050689) [inline src/syngen/ml/metrics/metrics_classes/metrics.py:1222]: Please. move it to the top of the file

File: src/syngen/ml/metrics/metrics_classes/metrics.py [hunk 2]
```diff
@@ -1145,35 +1169,69 @@ def __model_process(self, model_object, targets, task_type):
...
-            GradientBoostingClassifier(random_state=10),
+            RandomForestClassifier(n_jobs=-1, random_state=10),
             multiclass_targets,
             "multiclass classification",
         )
         return best_target, score, synthetic_score
 
     def __create_regression_models(self, cont_targets):
-        from sklearn.ensemble import GradientBoostingRegressor
+        from sklearn.ensemble import RandomForestRegressor
```
Comment (Anna050689) [inline src/syngen/ml/metrics/metrics_classes/metrics.py:1232]: Please. move it to the top of the file

**Review by Anna050689 — CHANGES_REQUESTED** [review]
